analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs

```diff
@@ -519,6 +521,53 @@ public void ReceivedInOrder()
         }
     }
 
+    // All assert methods start with "Should*". We do not test all APIs, just some.
+    public class ShouldlyTests
+    {
+        int i = 1;
+        public dynamic Foo() => null;
+        public void Bar() { }
+
+        [TestCase]
+        public void NoAssert() // Noncompliant
+        {
+        }
+
+        [TestCase]
+        public void W()
+        {
+            int[] empty = { };
+            empty.ShouldBeEmpty();
+
+            var dict = new Dictionary<string, string> { { "x", "x" } };
+            dict.ShouldContainKey("y");
+
+            Should.Throw<IndexOutOfRangeException>(() => Bar());
+        }
+
+        [TestCase]
+        public void X() => "Foo".ShouldMatchApproved();
```

Comment:
I am testing each because they are extension methods defined in separate static classes, so we need good coverage given we rely on the "Should" string and the logic may change in the future